core/maximal_repeats.py

```diff
@@ -110,25 +111,24 @@ def process(lcp_interval: Interval, suf_tab: list[int], bwt_table: list[str]) ->
             if len(prev_child_alphabet_post_sets.keys()) == 0:
                 prev_child_alphabet_post_sets = curr_child_alphabet_pos_sets
             else:
-                for curr_key in curr_child_alphabet_pos_sets:
-                    for prev_key in prev_child_alphabet_post_sets:
-                        if curr_key != prev_key:
-                            curr_alphabet_list_head = curr_child_alphabet_pos_sets[curr_key]
-                            prev_alphabet_list_head = prev_child_alphabet_post_sets[prev_key]
-                            curr_alphabet_list_ptr = curr_alphabet_list_head
+                for prev_key in prev_child_alphabet_post_sets:
+                    for curr_key in curr_child_alphabet_pos_sets:
+                        if prev_key != curr_key:
+                            prev_key_itr = prev_child_alphabet_post_sets[prev_key]
```

Comment:
Why is one named `_itr` and the other `itr_head`? Aren't they both just heads of a linked list?

You might consider wrapping them in proper iterators to make the code more Pythonic